### PR TITLE
Adding AWS Quickstart Terraform

### PR DESCRIPTION
**Pull request by Oats87**
Adding Terraform to support AWS Quickstarting

Updated README.md to reflect AWS Quick Start Usage instructions
Added AWS Terraform loosely based on DigitalOcean Terraform
Modified DO Cloud Init Scripts to perform AWS-specific functionality

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,3 +50,38 @@ When provisioning is finished the Rancher UI will become accessible on http://17
...
+
+You will need the following:
+
+- An AWS Account with an access key and secret key
+- The name of a pre-created AWS Key Pair
+- Your desired AWS Deployment Region
+
+This terraform setup will:
+
+- Start a droplet running `rancher/rancher` version specified in `rancher_version`
```
Comment (superseb) [inline README.md]: droplet -> instance

File: aws/files/userdata_agent [hunk 2]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash -x
+export curlprefix=appropriate
+export rancher_server_ip='${server_address}'
+
+INSTANCELOCALIP=`curl -s http://169.254.169.254/latest/meta-data/local-ipv4`
+INSTANCEPUBLICIP=`curl -s http://169.254.169.254/latest/meta-data/public-ipv4`
```
Comment (superseb) [inline aws/files/userdata_agent]: Both can be retrieved by passing either `awslocal` or `awspublic` as parameter to `--internal-address` and `--address`

File: aws/files/userdata_agent [hunk 3]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash -x
+export curlprefix=appropriate
```
Comment (chrisurwin) [inline aws/files/userdata_agent]: curlprefix is only required if using RancherOS

File: aws/main.tf [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+  description = "Cluster Prefix - All resources created by Terraform have this prefix prepended to them"
+}
+
+variable "rancher_version" {
+  default     = "v2.0.6"
+  description = "Rancher Server Version"
+}
+
+variable "count_agent_all_nodes" {
+  default     = "3"
```
Comment (chrisurwin) [inline aws/main.tf]: we are changing this to a default of 1 for all quickstarts, and instructions need to be updated to show how to increase this

File: aws/files/userdata_agent (hunk 3 above)
Comment (Oats87) [reply aws/files/userdata_agent]: There are parts of the userdata_agent script that utilize $curlprefix to pull the "appropriate/curl" image, should we hardcode that to "appropriate" instead?

**Comment by Oats87** [comment]
@superseb @chrisurwin Squashed commits, this should be ready to merge now

**Pull request merged**